presto-main/src/main/java/com/facebook/presto/type/DecimalSaturatedFloorCasts.java

```diff
@@ -95,4 +105,119 @@ private static BigInteger bigintToBigintFloorSaturatedCast(BigInteger value, int
         }
         return unscaledValue;
     }
+
+    public static final SqlScalarFunction DOUBLE_TO_DECIMAL_SATURATED_FLOOR_CAST = SqlScalarFunction.builder(DecimalSaturatedFloorCasts.class)
```

Comment:
> you could maybe use prestodb#5583, if it is in release branch

I would like to keep it as close to master as possible. FLOAT coercions should be merged soon. So this patch could be marged to master as well.

> what about DECIMAL_TO_DOUBLE_SATURATED_FLOOR_CAST?

There are no widening coercion from DOUBLE to DECIMAL.